document-generator-core/src/test/java/uk/gov/companieshouse/document/generator/core/document/DocumentGeneratorTest.java

```diff
@@ -0,0 +1,138 @@
+package uk.gov.companieshouse.document.generator.core.document;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.companieshouse.document.generator.core.avro.AvroDeserializer;
+import uk.gov.companieshouse.document.generator.core.document.models.RenderSubmittedDataDocument;
+import uk.gov.companieshouse.document.generator.core.kafka.ConsumerGroupHandler;
+import uk.gov.companieshouse.document.generator.interfaces.DocumentInfoService;
+import uk.gov.companieshouse.document.generator.interfaces.model.DocumentInfo;
+import uk.gov.companieshouse.kafka.consumer.CHKafkaConsumerGroup;
+import uk.gov.companieshouse.kafka.message.Message;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.UUID;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyList;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class DocumentGeneratorTest {
+
+    private static final String ID = UUID.randomUUID().toString();
+    private static final String TRANSACTIONS_RESOURCE = "transactions";
+    private static final String REQUESTER_ID = "requester-id";
+
+    private List<Message> messages;
+
+    private Message message;
+
+    private RenderSubmittedDataDocument renderSubmittedDataDocument;
+
+    private DocumentGenerator documentGenerator;
+
+    @Mock
+    private CHKafkaConsumerGroup mockConsumerGroup;
+
+    @Mock
+    private DocumentInfoService mockDocumentInfoService;
+
+    @Mock
+    private ConsumerGroupHandler mockConsumerGroupHandler;
+
+    @Mock
+    private AvroDeserializer<RenderSubmittedDataDocument> mockAvroDeserializer;
+
+    @BeforeEach
+    public void setUp() {
+
+        createTestMessageList();
+        createTestRenderSubmittedDataDocument();
+    }
+
+    @Test
+    @DisplayName("test the polling of Kafka and the consumer group commit")
+    public void testPollAndConsumerGroupCommit() throws IOException {
```

Comment:
You're missing a test for verifying a commit when an exception is caught

Reply:
added test to check